File: linked_list.py
import random

class Node:
    def __init__(self, val: int = 0):
        self.val = val
        self.next = None

class ListNode:
    def __init__(self, node: Node = None):
        self.head = node
        self.is_empty = True if node is None else False

    def insert_node(self, node: Node = None) -> bool:
        if node is None or not isinstance(node, Node):
            return False
        
        elif self.head is None:
            self.head = node
            self.is_empty = False
            return True

        else:
            curr = self.head
            while curr.next:
                curr = curr.next

            curr.next = node
            return True

    def print_list(self) -> bool:
        curr = self.head
        while curr is not None:
            print(curr.val)
            curr = curr.next
        return True

if __name__ == '__main__':
    list_node = ListNode()
    
    for i in range(10):
        n = random.randint(1, 10)
        print(f'The current number is {n}.')
        list_node.insert_node(Node(n))

    input('Inserting process completed. Press to continue: ')

    list_node.print_list()
    

File: linked_list.cpp
#include <iostream>
#include "linked_list"

using namespace std;

Node::Node(int val){
    this->val = val;
    this->next = NULL;
    return;
};

int Node::get_value(){
    return this->val;
};

Node* Node::get_next(){
    return this->next;
}

bool Node::set_next(Node* node){
    this->next = node;
    return true;
}

linked_list::linked_list(){
    this->head = NULL;
    this->is_empty = true;
};

linked_list::~linked_list(){
};

bool linked_list::insert_node(Node* node){
    if (this->is_empty){
        this->head = node;
        this->is_empty = false;
        return true;
    }
    else {
        Node* curr = this->head;
        while (curr->get_next())
            curr = curr->get_next();

        curr->set_next(node);
        return true;
    }
};

bool linked_list::print_list(){
    Node* curr = this->head;
    while (curr){
        cout << curr->get_value() << endl;
        curr = curr->get_next();
    }
    return true;
};